maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/MavenWorkingSessionImpl.java

```diff
@@ -188,6 +197,92 @@ public MavenWorkingSession loadPomFromFile(File pomFile, Properties userProperti
+                                                           Set<MavenDependency>  additionalDependencies) {
+        Collection<ArtifactResult> projectLocalDependencies = new ArrayList<>(depsForResolution.size());
+        for (MavenDependency dependency : depsForResolution) {
+            Path resolved = resolveProjectLocal(dependency.getGroupId(), dependency.getArtifactId(),
+                    dependency.getVersion(), Optional.ofNullable(dependency.getClassifier()),
+                    Optional.ofNullable(dependency.getPackaging().getExtension()), additionalDependencies);
+            if (resolved != null && resolved.toFile().exists()) {
+                Artifact artifact = new DefaultArtifact(dependency.getGroupId(), dependency.getArtifactId(),
+                        dependency.getClassifier(), dependency.getPackaging().getExtension(), dependency.getVersion(),
+                        null, resolved.toFile());
+                ArtifactResult result = new ArtifactResult(new ArtifactRequest()
+                        .setDependencyNode(new DefaultDependencyNode(
+                                new Dependency(artifact, dependency.getScope().name(), dependency.isOptional()))));
+                result.setArtifact(artifact);
+                projectLocalDependencies.add(result);
+            }
+        }
+        return projectLocalDependencies;
+    }
+
+    private Path resolveProjectLocal(String groupId, String artifactId, String version,
+                                     Optional<String> classifier, Optional<String> extension,
+                                     Set<MavenDependency> additionalDependencies) {
+        Path projectLocalRepository = findProjectLocalRepository();
+        if (projectLocalRepository == null) {
+            return null;
+        }
+
+        Predicate<String> isNotEmpty = s -> !s.isEmpty();
+        processAdditionalDependencies(projectLocalRepository, groupId, artifactId, version,
+                additionalDependencies);
+
+        return projectLocalRepository.resolve(groupId).resolve(artifactId).resolve(version)
+                .resolve(toVersionedArtifact(artifactId, version)
+                        + classifier.filter(isNotEmpty).map(c -> "-" + c).orElse("")
+                        + "." + extension.filter(isNotEmpty).orElse("jar"));
+    }
+
+    private static String toVersionedArtifact(String artifactId, String version) {
+        return artifactId + "-" + version;
+    }
+
+    private void processAdditionalDependencies(Path projectLocalRepository, String groupId,
+                                               String artifactId, String version,
+                                               Set<MavenDependency> additionalDependencies) {
+        Path directory = projectLocalRepository.resolve(groupId).resolve(artifactId).resolve(version);
+        File consumerPom = directory.resolve(toVersionedArtifact(artifactId, version) + "-consumer.pom").toFile();
```

Comment:
This is Maven internal thing, what happens if layout or naming is modified? This is -1